pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_sni.go

```diff
@@ -219,6 +253,84 @@ func (s *sni) emptyVirtualService() *istionetworkingv1beta1.VirtualService {
 	return &istionetworkingv1beta1.VirtualService{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeAPIServer, Namespace: s.namespace}}
 }
 
+func (s *sni) emptyAuthorizationPolicy(name string) *istiosecurity1beta1.AuthorizationPolicy {
+	return &istiosecurity1beta1.AuthorizationPolicy{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: s.values.IstioIngressGateway.Namespace}}
+}
+
+func (s *sni) emptyAuthorizationPolicyApiServer() *istiosecurity1beta1.AuthorizationPolicy {
+	return s.emptyAuthorizationPolicy(s.namespace + "-api-server")
+}
+
+func (s *sni) emptyAuthorizationPolicyVpnServer() *istiosecurity1beta1.AuthorizationPolicy {
+	return s.emptyAuthorizationPolicy(s.namespace + "-vpn-server")
+}
+
+func (s *sni) getAccessControlSpec() (*istioapisecurityv1beta1.AuthorizationPolicy, error) {
+	var err error
+	action := istioapisecurityv1beta1.AuthorizationPolicy_ALLOW
+	rules := []*istioapisecurityv1beta1.Rule{{From: []*istioapisecurityv1beta1.Rule_From{}}}
+
+	if s.values.AccessControl != nil {
+		ac := s.values.AccessControl
+		action, err = toIstioAuthPolicyAction(ac.Action)
+		if err != nil {
+			return nil, err
+		}
+
+		rules = []*istioapisecurityv1beta1.Rule{{
+			From: []*istioapisecurityv1beta1.Rule_From{{
+				Source: &istioapisecurityv1beta1.Source{
+					IpBlocks:          notNilSlice(ac.Source.IPBlocks),
```

Comment:
The `*Ip*` fields in `istioapisecurityv1beta1.Source` should be optional. Is it a problem if we pass nil slices here?

Reply:
The problem we had when we just pass nil slices, that the mutation func in `GetAndCreateOrMergePatch` does not remove existing blocks when a user updates the shoot resources from a number of blocks to none.